### PR TITLE
Enable inference_microbenchmark_smoke_test

### PR DESCRIPTION
**Pull request by SurbhiJainUSC**
# Description
This PR enables `inference_microbenchmark_smoke_test` which was disabled during linting effort [here](https://github.com/AI-Hypercomputer/maxtext/pull/1482).

*Notice 1:* Once all tests pass, the "pull ready" label will automatically be assigned.
This label is used for administrative purposes. Please do not add it manually.

*Notice 2:* For external contributions, our settings currently require an approval from a MaxText maintainer to trigger CI tests.

# Tests
`pytest MaxText/tests/integration_tests/inference_microbenchmark_smoke_test.py` 

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Review by khatwanimohit — APPROVED** [review]

**Review by bvandermoon — APPROVED** [review]

File: MaxText/prefill_packing.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def _process_compiled(self, params: Params, padded_length: int):
       self.process_func[padded_length] = (
           jax.jit(
               self._process,
-              in_shardings=(self.engine.param_layouts, None, None, None, self.engine.decode_state_layouts),
+              in_shardings=(self.engine.param_layouts, None, None, None, self.engine.decode_state_layouts, None),
```
[the remark this answers is not in the document]
Comment (SurbhiJainUSC) [reply MaxText/prefill_packing.py:131]: It is for `rng` at line 144.
